fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
```

Comment:
If someone is able to use the above example of a configuration and show how modules can expose a various number of strongly typed components, then it's ok that `Constructor` takes no arguments.

> Also this contradicts with what you said earlier Glib about Constructor taking no args

Yes, I know that I've previously argued for it taking no arguments, but the configuration is very special, even more so than the tracing, logging and metrics.